packages/http/src/router/matchBaseUrl.ts

```diff
@@ -1,23 +1,55 @@
+import * as E from 'fp-ts/lib/Either';
+import * as O from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+
 import { INodeVariable, IServer } from '@stoplight/types';
 import { MatchType } from './types';
 
 const variableRegexp = /{(.*?)}/g;
 
-export function matchBaseUrl(server: IServer, baseUrl: string): MatchType {
-  const templateMatchResult = matchRequestUrlToTemplateUrl(baseUrl, server.url, server.variables);
+export function matchBaseUrl(server: IServer, baseUrl: string) {
+  return pipe(
+    convertTemplateToRegExp(server.url, server.variables),
+    E.map(regex =>
+      pipe(
+        O.fromNullable(regex.exec(baseUrl)),
```

Comment:
Either has a `fromNullable` method as well — and then you can use `E.orElse` to return a fallback value so that it gets transformed in a `right`

Reply:
I'll try that, thanks for the tip.